asreview/datasets.py

```diff
@@ -29,203 +29,195 @@ class DataSetNotFoundError(Exception):
+            'topic': self.topic,
+            'link': self.link,
+            'reference': self.reference,
+            'license': self.license,
+            'year': self.year,
+            'img_url': self.img_url,
+            'aliases': self.aliases,
+            **self.kwargs
+        }
 
-    def list(self, latest_only=True):
-        """Return a list of itself."""
-        return [self]
 
+class BaseDataGroup():
+    def __init__(self, *datasets):
+        """Group of datasets.
 
-class BaseVersionedDataSet():
-    def __init__(self, base_dataset_id, datasets, title="Unknown title"):
-        """Initialize a BaseVersionedDataDet instance.
+        Group containing one or more datasets.
 
         Parameters
         ----------
-        base_dataset_id: str
-            This is the identification given to the versioned dataset as a
-            whole. Each individual version has their own dataset_id.
-        datasets: list
-            List of BaseDataSet's containing all the different versions.
+        *datasets:
+            One or more datasets.
         """
-        self.datasets = datasets
-        self.title = title
-        self.dataset_id = base_dataset_id
+        self.datasets = list(datasets)
 
     def __str__(self):
-        dataset_dict = self.datasets[-1].to_dict()
-        dataset_dict["versions_available"] = [
-            d.dataset_id for d in self.datasets
-        ]
-        return pretty_format(dataset_dict)
-
-    def __len__(self):
-        return len(self.datasets)
+        return f"<BaseDataGroup group_id='{self.group_id}'>"
 
-    def find(self, dataset_name):
-        if dataset_name == self.dataset_id:
-            return self
-
-        all_dataset_names = [(d, d.aliases) for d in self.datasets]
-        for dataset, aliases in all_dataset_names:
-            if dataset_name.lower() in aliases:
-                return dataset
-
-        raise DataSetNotFoundError(
-            f"Dataset {dataset_name} not found"
-        )
+    def __dict__(self):
+        return {d.dataset_id: d for d in self.datasets}
 
-    def get(self, i_version=-1):
-        return self.datasets[i_version].get()
-
-    def list(self, latest_only=True):
-        if latest_only:
-            return [self.datasets[-1]]
-        return [self]
-
-
-class BaseDataGroup():
-    def __init__(self, *args):
-        """Group of datasets."""
-        self._data_sets = [a for a in args]
+    def append(self, dataset):
+        """Append dataset to group.
 
-    def __str__(self):
-        return "".join([
-            f"*******  {str(data.dataset_id)}  *******\n"
-            f"{str(data)}\n\n" for data in self._data_sets
-        ])
+        dataset: asreview.datasets.BaseDataSet
+            A asreview BaseDataSet-like object.
+        """
+        if not issubclass(dataset, BaseDataSet):
+            raise ValueError(
+                "Expected BaseDataSet or subclass of BaseDataSet."
+            )
+        self.datasets.append(dataset)
 
```

Comment:
issubclass() checks whether the argument is a class, not an instance of a class. This means that appending a dataset is not possible:

![image](https://user-images.githubusercontent.com/28191416/145820035-67d49a12-468b-48a9-a6f4-7de10273630a.png)

![image](https://user-images.githubusercontent.com/28191416/145820068-726aa697-c7cc-4c97-8d08-bafc96c9166b.png)

Reply:
Maybe I am using the function wrong! I should be giving classes. One second, let me test this.